packages/bootstrap-chat/README.md

```diff
@@ -0,0 +1,141 @@
+# bootstrap-chat
+
+A Bootstrap-based set of UI components for facilitating chat-based tasks for Mephisto.
+
+## Installation
+
+```bash
+npm install --save bootstrap-chat mephisto-task
+```
+
+Install from local folder (e.g. for local development):
+
+```bash
+cd packages/bootstrap-chat
+npm link
+
+cd <app folder>
+npm link bootstrap-chat
+
+# If you're getting an invalid hooks error (https://reactjs.org/warnings/invalid-hook-call-warning.html),
+# you can also do the following to ensure that both the app
+# and bootstrap-chat are using the same version of React:
+# 
+# cd packages/bootstrap-chat
+# npm link ../<app folder>/node_modules/react
+
+```
+
+## Usage
+
+This package contains both individual components for building a chat UI, as well as precomposed components that can get you going quickly.
+
+We generally recommend starting with the precomposed components, unless you have an advanced use-case. We provide escape-hatches for easy customization of common behaviors.
+
+### Example 1: The pre-composed ChatApp component
+
+The most simplest way to get going with 0 customizations is just the following:
+
+```jsx
+import { ChatApp } from "bootstrap-chat";
+
+ReactDOM.render(<ChatApp />, document.getElementById("root"));
+```
+
+### Example 2: Modifying ChatApp
+
+From here, you can easily opt into adding custom components and behaviors via render prop overrides.
+
+Let's consider a contrived example where we would like every message to appear suffixed with 5 exclamation marks.
+
+```jsx
+
+import { ChatApp, ChatMessage } from "bootstrap-chat";
+
+function MyChatApp() {
+    return (
+        <ChatApp
+            renderMessage={({ message }) => <ChatMessage message={message.text + "!!!!!"} />}
+        />);
+}
+
+```
+
+Here we replace the rendering of the chat message with the `renderMessage` prop. However, note that we also import the `<ChatMessage />` component and use that as part of the rendering. We don't have to do this portion, and could easily just create our own isolated implementation. However this example illustrates how we can opt out of default behavior, while still having available to us primitives and core components that can get us back some of that default functionality at our discretion.
+
+
+## `<ChatApp />` documentation
+
+`ChatApp` exposes the following props: 
+
+All render props will receive as part of their args both `mephistoContext` and `appContext`, which will be described below.
+
+
+### `renderMessage({ message, idx, mephistoContext, appContext })`
+A render prop that if implemented, overides the way a chat message is rendered.
+
+**Arguments:**
+
+**`message`**: An object with the following schema:
+
+TODO: where does this object come from in the backend? Find to verify the below
```

Comment:
Want to flag this question - where exactly does the `message` object schema get defined in the backend?

Reply:
This actually comes all the way from the ParlAI definition of the [`Message`](https://github.com/facebookresearch/ParlAI/blob/bca1dd153845c265f9ce53fe0ff0ab51666c878d/parlai/core/message.py#L14) class, but I'm realizing now that that no longer explicitly sets certain fields. We should probably standardize some keys that we expect here though:

- `'id'`: actually the speaker id for the message, but ParlAI has called this the id
- `'message_id'`: We add this when a message is passed to ensure it's always unique, no funny business.
- `'text'`: This contains the actual message text, and the message should not be displayed if this is undefined or the empty string
- `'task_data'`: This is the dict that is used to update the `taskContext` variable, and is later accessible from the message for specific messages.

Reply:
Thanks! Added this info.